File: modelGuide/src/main/java/com/wynne/knowledge/guide/material/RecycleActivity.java
package com.wynne.knowledge.guide.material;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;


import com.wynne.knowledge.base.base.BaseActivity;
import com.wynne.knowledge.guide.R;

import java.util.Arrays;
import java.util.List;

/**
 * @author Wynne
 * @date 2018/4/27
 */

public class RecycleActivity extends BaseActivity {
    private String[] arrys = {
            "Item1", "Item2", "Item3", "Item4",
            "Item5", "Item6", "Item7", "Item8",
            "Item9", "Item10", "Item11", "Item12",
            "Item13", "Item14", "Item15", "Item16",
            "Item17", "Item18", "Item19", "Item20"
    };
    private List<String> mList;


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.mine_layout);
        initView();

    }

    @Override
    public void initView() {
        mList = Arrays.asList(arrys);
        RecyclerView recyclerView = findViewById(R.id.rv_list);
        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);
        RecycleViewAdapter adapter = new RecycleViewAdapter(this);
        recyclerView.setAdapter(adapter);
    }


    @Override
    public int getLayoutId() {
        return R.layout.mine_layout;
    }


    public class RecycleViewAdapter extends RecyclerView.Adapter<RecycleViewAdapter.ViewHolder> {
        private Context mContext;

        public RecycleViewAdapter(Context context) {
            mContext = context;
        }

        @Override
        public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            Log.d("XXW", "viewType :" + viewType);
            View view = LayoutInflater.from(mContext).inflate(R.layout.layout_item, parent, false);
            return new ViewHolder(view);
        }

        @Override
        public void onBindViewHolder(ViewHolder holder, int position) {
            holder.mTextView.setText(mList.get(position) + " === " + position);
        }

        @Override
        public int getItemCount() {
            return mList.size();
        }

        class ViewHolder extends RecyclerView.ViewHolder {
            private TextView mTextView;

            public ViewHolder(View itemView) {
                super(itemView);
                mTextView = (TextView) itemView.findViewById(R.id.tv_item);
            }
        }


    }
}

File: modelGuide/src/main/java/com/wynne/knowledge/guide/material/RecycleActivity.kt
package com.wynne.knowledge.guide.material

import android.content.Context
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView


import com.wynne.knowledge.base.base.BaseActivity
import com.wynne.knowledge.guide.R

import java.util.Arrays

/**
 * @author Wynne
 * @date 2018/4/27
 */

class RecycleActivity : BaseActivity() {
    private val arrys = arrayOf("Item1", "Item2", "Item3", "Item4", "Item5", "Item6", "Item7", "Item8", "Item9", "Item10", "Item11", "Item12", "Item13", "Item14", "Item15", "Item16", "Item17", "Item18", "Item19", "Item20")
    private var mList: List<String>? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.mine_layout)
        initView()

    }

    override fun initView() {
        mList = Arrays.asList(*arrys)
        val recyclerView = findViewById<RecyclerView>(R.id.rv_list)
        val layoutManager = LinearLayoutManager(this)
        recyclerView.layoutManager = layoutManager
        val adapter = RecycleViewAdapter(this)
        recyclerView.adapter = adapter


    }


    override fun getLayoutId(): Int {
        return R.layout.mine_layout
    }


    inner class RecycleViewAdapter(private val mContext: Context) : RecyclerView.Adapter<RecycleViewAdapter.ViewHolder>() {

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
            Log.d("XXW", "viewType :$viewType")
            val view = LayoutInflater.from(mContext).inflate(R.layout.layout_item, parent, false)
            return ViewHolder(view)
        }

        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
            holder.mTextView.text = mList!![position] + " === " + position
        }

        override fun getItemCount(): Int {
            return mList!!.size
        }

        inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
            val mTextView: TextView

            init {
                mTextView = itemView.findViewById<View>(R.id.tv_item) as TextView
            }
        }


    }
}
